transport/src/iosMain/kotlin/com/genesys/cloud/messenger/transport/network/PlatformSocket.kt

```diff
@@ -1,60 +1,123 @@
 package com.genesys.cloud.messenger.transport.network
 
-import cocoapods.jetfire.JFRWebSocket
 import com.genesys.cloud.messenger.transport.core.Configuration
+import com.genesys.cloud.messenger.transport.util.extensions.string
+import com.genesys.cloud.messenger.transport.util.extensions.toNSData
 import com.genesys.cloud.messenger.transport.util.logs.Log
+import kotlinx.cinterop.convert
 import platform.Foundation.NSData
+import platform.Foundation.NSError
+import platform.Foundation.NSMutableURLRequest
+import platform.Foundation.NSOperationQueue
+import platform.Foundation.NSPOSIXErrorDomain
 import platform.Foundation.NSTimer
 import platform.Foundation.NSURL
+import platform.Foundation.NSURLSession
+import platform.Foundation.NSURLSessionConfiguration
+import platform.Foundation.NSURLSessionWebSocketCloseCode
+import platform.Foundation.NSURLSessionWebSocketDelegateProtocol
+import platform.Foundation.NSURLSessionWebSocketMessage
+import platform.Foundation.NSURLSessionWebSocketTask
+import platform.Foundation.setValue
+import platform.darwin.NSObject
+import platform.posix.ETIMEDOUT
 
 internal actual class PlatformSocket actual constructor(
     private val log: Log,
     configuration: Configuration,
+    /**
+     * Interval to automatically send pings while active. If pong not received within `interval`,
+     * client assumes connectivity is lost and will notify [PlatformSocketListener.onFailure].
+     */
```

Comment:
Is that something we would want to configure?

Reply:
Do you mean allow the configuring of whether `onFailure` is notified if pong not received within `interval`? I set it up this way to more closely match how OkHttp works on the Android side which I figured would be sufficient.